### DISCUSSION TITLE
Trouble upgrading to new version

### DISCUSSION THREAD

**Discussion by un99known99** [PrivateBin/PrivateBin#918] [General]
Hi, I am currently running 1.3.5 on up-to-date debian bullseye.
I tried to follow the instructions to upgrade (extract and copy over the packed content).

Either I am doing that wrong, but changed files are not going to be updated.

May you could share the commands you used to upgrade (I am not an expert) to 1.4.0. 

Would be also help to add some more technical infos about how to upgrade in the release notes.
Looking forward to you answer.
Many thanks,
Mike

**Comment by elrido** [comment]
On the demo site we use the docker container, so the upgrade was just deploying the 1.4.0 image over the 1.3.5 one. If you still see the old version in the browser, after replacing the files on disk, I suggest checking your webserver and/or CDN configuration for any cache related settings that may cause the old version to still get used.

**Comment by un99known99** [comment]
hm, I have a single instance on a pi, no docker or similar.
I tried unpacking the new version to a folder then copy the contenct of the new folder to /var/www/html/PrivateBin/*, even though I did it as root files with changed timestamp (new version) got not reflected in existing installation.

Is there a different method in upgrading rather then "cp -rp NEW140/* OLD135/*" and restarting apache2? 
Above was not working for me - any help?

**Reply by elrido** [reply]
There should really not be anything more complicated needed. Even restarting the server should not be necessary. If you can just place the new installation into a separate folder you could verify if there is any caching going on that shows you incorrect data.

**Comment by un99known99** [comment]
Thx, the upgrade with merging 1.4.0 into 1.3.5 worked.
But in release notes I didnt find any *instructions* on what to change in config file for all the new stuff - besides the header recomms (which I implemented), news are referenced to open issues but no option set something in config file or similar is shown, any hint here?

**Reply by elrido** [reply]
You should not have to change your config file during upgrades, except if you want to disable or change any of the new settings.

Ideally, your `conf.php` only contains the settings that are different from the defaults shown and explained in `conf.sample.php`. Further information can be found in the [wiki article on configuration](https://github.com/PrivateBin/PrivateBin/wiki/Configuration).

**Comment by officialspitxxx-creator** [comment]
Hola, quiero escribir blogs para mi sitio web https://spitxxx.com También puedo pagar por blogs de buena calidad. Si necesitas ayuda, avísame.
